Consistency between Span and Resource attributes (#777)
open-telemetry/opentelemetry-specification: specification/common/common.md

    i.e. it MUST NOT contain values of different types. For protocols that do
    not natively support array values such values SHOULD be represented as JSON strings.

Attribute values expressing a numerical value of zero, an empty string, or an
empty array are considered meaningful and MUST be stored and passed on to
processors / exporters. Attribute values of `null` are considered to be not set

> and get discarded as if that `Attribute` was never been created.

Reviewer: ```suggestion
and get discarded as if that `Attribute` has never been created.
```
or
```suggestion
and get discarded as if that `Attribute` was never created.
```

> As an exception to this, if overwriting of values is supported, this results in

Reviewer 2: It's not really an exception to the rule above but rather a consequence of it.

Reviewer 3: I think it's an exception. The normal consequence would IMHO be to just ignore all calls to set with a `null` argument, meaning after `span.SetAttribute('foo', 'bar'); span.SetAttribute('foo', null);`, the `foo` attribute has the value `bar`. I wonder if we should change the behavior, but this would be outside the scope of this PR.

Reviewer 4: Based on this idea, would like to see an issue with this to be finalized :) As you said a separate PR.

Author: I created #797 to discuss this

> As an exception to this, if overwriting of values is supported, this results in
> clearing the previous value and dropping the attribute key from the set of attributes.

Reviewer 3: Why do we say both "clearing the previous value *and* dropping the attribue key"? Would it not be more clear to just say

```suggestion
As an exception to this, if overwriting of values is supported, this removes the attribute.
```

Author: I kept the same language that was already used, but I think your suggestion makes the text more clear.

Author: done in 73e79fd
